packages/scm/src/browser/scm-provider.ts

```diff
@@ -69,11 +69,23 @@ export interface ScmCommand {
 }
 
 export interface ScmCommit {
-    id: string,  // eg Git sha or Mercurial revision number
-    summary: string,
-    authorName: string,
-    authorEmail: string,
-    authorDateRelative: string
+    readonly id: string;  // eg Git sha or Mercurial revision number
+    readonly commitDetailUri: URI;
+    readonly summary: string;
+    readonly authorName: string;
+    readonly authorEmail: string;
+    readonly authorDateRelative: string;
+    readonly fileChanges: ScmFileChange[];
+    readonly commitDetailOptions: {};
```

Comment:
`scm-extra` and `git` extension are very likely going to be moved to a separate repo soon and replace by vscode `git` and `git-ui` extensions.

There should be a way that `git` and `scm-extra` can add new properties without adding them to `scm` package. `scm` package should contain only common part between features needed to support VS Code extensions and scm-extra but not more.

Reply:
I've moved fileChanges and commitDetailOptions out of scm package, also ScmFileChange.  ScmCommit remains for time being as it is used by 'amend' support.